Composer/packages/tools/language-servers/language-generation/src/LGServer.ts

```diff
@@ -173,60 +166,33 @@ export class LGServer {
     );
   }
 
-  protected getImportResolver(document: TextDocument) {
-    const editorContent = document.getText();
-    const internalImportResolver = () => {
-      return {
-        id: document.uri,
-        content: editorContent,
-      };
-    };
-    const { fileId, templateId, projectId } = this.getLGDocument(document) || {};
-
-    if (this.importResolver && fileId && projectId) {
-      const resolver = this.importResolver;
-      return (source: string, id: string) => {
-        const lgFile = resolver(source, id, projectId);
-        if (!lgFile) {
-          this.sendDiagnostics(document, [
-            generageDiagnostic(`lg file: ${fileId}.lg not exist on server`, DiagnosticSeverity.Error, document),
-          ]);
-        }
-        let { content } = lgFile;
-        /**
-         * source is . means use as file resolver, not import resolver
-         * if inline editor, server file write may have delay than webSocket updated LSP server
-         * so here build the full content from server file content and editor content
-         */
-        if (source === '.' && templateId) {
-          content = updateTemplate(lgFile.content, templateId, editorContent);
-        }
-        return { id, content };
-      };
-    }
-
-    return internalImportResolver;
-  }
-
   protected addLGDocument(document: TextDocument, lgOption?: LGOption) {
     const { uri } = document;
     const { fileId, templateId, projectId } = lgOption || {};
-    const index = () => {
-      const importResolver: ImportResolverDelegate = this.getImportResolver(document);
+    const index = async () => {
       let content: string = document.getText();
       // if inline mode, composite local with server resolved file.
-      if (this.importResolver && fileId && templateId) {
+      if (this.getLgResources && fileId && templateId) {
         try {
-          content = importResolver('.', `${fileId}.lg`).content;
+          const resources = this.getLgResources(projectId);
+          content = resources.find((item) => item.id === fileId)?.content || content;
         } catch (error) {
           // ignore if file not exist
         }
       }
 
       const id = fileId || uri;
-      const { allTemplates, diagnostics } = Templates.parseText(content, id, importResolver);
```

Comment:
Here we still need `allTemplates` for intelligent suggestions (items in imported file)